lib/windows.js

```diff
@@ -119,9 +124,58 @@ function windows() {
 	// Remove null characters from buffer
 	const processFileNameBufferClean = ref.reinterpretUntilZeros(processFileNameBuffer, wchar.size);
 	// Get process file path as a string
-	const processPath = wchar.toString(processFileNameBufferClean);
+	let processPath = wchar.toString(processFileNameBufferClean);
 	// Get process file name from path
-	const processName = path.basename(processPath);
+	let processName = path.basename(processPath);
+	let newProcessName = null;
+	let newProcessPath = null;
+
+	// ApplicationFrameHost & Universal Windows Platform Support
+	if (processName === 'ApplicationFrameHost.exe') {
+		const windowProc = ffi.Callback('bool', ['pointer', 'int32'], function (hwnd, lParam) {
+			// Allocate a buffer to store the process ID
+			const processIdBufferChild = ref.alloc('uint32');
+			// Write the process ID creating the window to the buffer (it returns the thread ID, but it's not used here)
+			user32.GetWindowThreadProcessId(hwnd, processIdBufferChild);
+			// Get the process ID as a number from the buffer
+			const processIdChild = ref.get(processIdBufferChild);
+			// Get a "handle" of the process
+			const processHandleChild = kernel32.OpenProcess(PROCESS_QUERY_LIMITED_INFORMATION, false, processIdChild);
+			if (ref.isNull(processHandleChild)) {
+				return undefined; // Failed to get process handle
+			}
+
+			// Set the path length to more than the Windows extended-length MAX_PATH length
+			const pathLengthBytesChild = 66000;
+			// Path length in "characters"
+			const pathLengthCharsChild = Math.floor(pathLengthBytesChild / 2);
+			// Allocate a buffer to store the path of the process
+			const processFileNameBufferChild = Buffer.alloc(pathLengthBytesChild);
+			// Create a buffer containing the allocated size for the path, as a buffer as it must be writable
+			const processFileNameSizeBufferChild = ref.alloc('uint32', pathLengthCharsChild);
+			// Write process file path to buffer
+			kernel32.QueryFullProcessImageNameW(processHandleChild, 0, processFileNameBufferChild, processFileNameSizeBufferChild);
+			// Remove null characters from buffer
+			const processFileNameBufferCleanChild = ref.reinterpretUntilZeros(processFileNameBufferChild, wchar.size);
+			// Get process file path as a string
+			const processPathChild = wchar.toString(processFileNameBufferCleanChild);
+			// Get process file name from path
+			const processNameChild = path.basename(processPathChild);
+
+			if (processNameChild !== processName) {
+				newProcessName = processNameChild;
+				newProcessPath = processPathChild;
+			}
+			return true;
+		});
+		user32.EnumChildWindows(activeWindowHandle, windowProc, 0);
```

Comment:
The logic here should be moved to a separate function that returns the name and path.

Reply:
I'll make a recommendation to consolidate the duplication of code and introduce functions that gather the `processId, processHandle` and a secondary function which returns `processPath, processName`, however since this callback is called multiple times (since there are many sub windows in a UWP), I'm not entirely sure how I'd structure this in such a way that can make the callback return the name and path, then process.

Reply:
I won't resolve this till feedback @sindresorhus